### ISSUE TITLE
Match iPadOS 13.4 mouse/trackpad cursor behavior

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#52912]
iPadOS 13.4 supports mouse/trackpad for 2018 iPad Pro.  However, it doesn't look like a normal desktop-like cursor and highlights differently on hover, and probably other behavior differences.
See video: https://twitter.com/bzamayo/status/1240255536601485312

**Comment by jmagman** [comment]
Blocked by https://github.com/flutter/flutter/issues/23604.

**Comment by jmagman** [comment]
Not sure if Catalyst will have something similar for macOS.

**Comment by bernaferrari** [comment]
From their reasoning when making this behavior (specially since it originated from the accessibility team), it seems it will be limited to iPad. I don't think they are planning to replace the macOS cursor with a circle.

**Comment by dancojocaru2000** [comment]
> Blocked by #23604.

Why? I don't know the technical details, but it looks to me like Flutter needs to report to iOS what size and shape the hovered element has in order for the cursor to change to that shape. That would be the most basic implementation, without things like FABs enlarging themselves and the cursor hiding.

**Comment by xster** [comment]
Design doc https://developer.apple.com/design/human-interface-guidelines/ios/user-interaction/pointers/ 
API docs https://developer.apple.com/documentation/uikit/pointer_interactions

**Comment by xster** [comment]
Though I don't know how we can test this. The keyboard trackpad isn't selling yet and 3rd gen iPad Pro simulator on macOS with magic trackpad doesn't simulate it 🤷‍♂️

**Comment by dancojocaru2000** [comment]
On a physical device, any Bluetooth or USB mouse works. Not sure about simulators though.

**Comment by bernaferrari** [comment]
@xster any mouse in a physical device should do it. It works with both bluetooth and wired mouses/trackpads.

**Comment by xster** [comment]
awesome, thanks for the pro tips. Updating my ipad...

**Comment by xster** [comment]
Not too bad. Trying on flutter/gallery, more things work than doesn't work among existing functionalities. 
#53453 is the most noticeable bug. 

Otherwise, we need to support the new features such as cursors, snaps, hovers etc, but since we don't fully support iPad form factors for tablet specific features, it's lower priority.

**Comment by jans-y** [comment]
<img width="1046" alt="image" src="https://user-images.githubusercontent.com/21024157/77821473-d798e080-70ea-11ea-9d71-90202f7a0383.png">

**Comment by Kavantix** [comment]
The scroll wheel also doesnt work.
I opened an issue for it: flutter/flutter#54663

**Comment by neckaros** [comment]
> but since we don't fully support iPad form factors for tablet specific features, it's lower priority.

Not sure what you mean by that? How can iPad not be considered a tablet? Or did I misunderstood you?

**Comment by dkwingsmt** [comment]
I've created an umbrella issue that tracks the support for iPadOS pointer interactions: https://github.com/flutter/flutter/issues/55809.
However, features beyond basic cursor type and hover effect will probably have a low priority for a while.

**Comment by dkwingsmt** [comment]
I'll close this issue in favor of https://github.com/flutter/flutter/issues/55809. Feel free to comment and like that issue :)
